### PR TITLE
rabbit holes

### PR DESCRIPTION
**Pull request by dwick**
:eyeglasses: @danehansen @curioussavage 

Got sucked into a rabbit hole while working on the error stuff, there's a few random commits in here too.

relies on https://github.com/reddit/horse/pull/5

### CODE REVIEW

File: src/lib/savedReply.es6.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+'use strict';
```
Comment (dwick) [inline src/lib/savedReply.es6.js]: we only save 1 comment replies so i've stored it on a single key, this just wraps up the behavior for multiple components to use

File: src/views/components/Comment.jsx [hunk 2]
```diff
@@ -37,15 +37,15 @@ class Comment extends BaseComponent {
...
-        savedReply: savedReply,
-        showReplyBox: true,
-        showTools: true,
+        showReplyBox: hasSavedReply,
+        showTools: hasSavedReply,
       });
-      var domNode = React.findDOMNode(this);
-      domNode.scrollIntoView();
+
+      React.findDOMNode(this).scrollIntoView();
```
Comment (dwick) [inline src/views/components/Comment.jsx]: `React.findDOMNode` is deprecated

File: src/views/components/Comment.jsx [hunk 3]
```diff
@@ -224,9 +224,6 @@ class Comment extends BaseComponent {
       highlighted = 'comment-highlighted';
 
       if (this.state.showReplyBox) {
-        if (this.state.savedReply) {
```
Comment (dwick) [inline src/views/components/Comment.jsx]: copying things between props/state and vis versa is an anti pattern

File: src/views/components/CommentBox.jsx [hunk 4]
```diff
@@ -10,76 +11,66 @@ class CommentBox extends BaseComponent {
...
   }
 
   handleInputChange (e) {
-    this.setState({savedReply: this.refs.text.value})
-    var textEl = this.refs.text.getDOMNode();
+    var el = React.findDOMNode(this.refs.text);
+    var value = el.value;
+
     this.setState({
-      inputCssClass: textEl.value.trim().length ? 'has-content' : '',
```
Comment (dwick) [inline src/views/components/CommentBox.jsx]: state that relies on other state is hard to keep in sync and should just be calculated on `render`

File: src/views/components/CommentBox.jsx [hunk 5]
```diff
@@ -10,76 +11,66 @@ class CommentBox extends BaseComponent {
...
 
     this.state = {
-      inputCssClass: '',
-      savedReply: props.savedReply || '',
+      reply: savedReply.get(this.props.thingId),
     };
   }
 
   handleInputChange (e) {
-    this.setState({savedReply: this.refs.text.value})
```
Comment (dwick) [inline src/views/components/CommentBox.jsx]: this didn't do anything

File: src/views/components/CommentBox.jsx [hunk 6]
```diff
@@ -10,76 +11,66 @@ class CommentBox extends BaseComponent {
...
+    var value = el.value;
+
     this.setState({
-      inputCssClass: textEl.value.trim().length ? 'has-content' : '',
+      reply: value,
     });
-  }
-
-  componentDidMount () {
-    if (this.props.savedReply) {
```
Comment (dwick) [inline src/views/components/CommentBox.jsx]: no longer relying on this being passed in

File: src/views/components/CommentBox.jsx [hunk 7]
```diff
@@ -10,76 +11,66 @@ class CommentBox extends BaseComponent {
...
-      this.setState({inputCssClass: 'has-content'});
-    }
-  }
-
-  componentWillUnmount () {
-    window.localStorage.clear();
```
Comment (dwick) [inline src/views/components/CommentBox.jsx]: this was clobbering anything we had in local storage when the component unmounted.  seemed undesirable

File: src/views/components/CommentBox.jsx [hunk 8]
```diff
@@ -10,76 +11,66 @@ class CommentBox extends BaseComponent {
...
-  }
 
-  componentWillReceiveProps (nextProps) {
-    this.setState({savedReply: nextProps.savedReply})
+    savedReply.set(this.props.thingId, value);
```
Comment (dwick) [inline src/views/components/CommentBox.jsx]: saving on update instead of just submit now

**Comment by danehansen** [comment]
this looks :fish: to me, but probably @curioussavage should check it out too since i think he wrote the messages and comments stuff.

**Comment by curioussavage** [comment]
im getting `Uncaught TypeError: Cannot read property 'thingId' of null` on all /comments route ex: http://localhost:4444/r/pics/comments/3fm8tz/robert_downey_jr_in_salt_and_pepper/

**Comment by curioussavage** [comment]
:haircut:

**Comment by dwick** [comment]
:haircut: need to do some rebasing after review

**Comment by curioussavage** [comment]
:haircut:

**Comment by curioussavage** [comment]
this is :fish: for me

**Pull request merged**